lib/wit/gitrepo.py

```diff
@@ -242,14 +246,91 @@ def is_ancestor(self, ancestor, current=None):
         return proc.returncode == 0
 
     def repo_entries_from_commit(self, revision) -> List[RepoEntry]:
-        where = "{}:{}".format(revision, GitRepo.PKG_DEPENDENCY_FILE)
-        proc = self._git_command("show", where)
+        manifest_entries = self._read_manifest_from_commit(revision)
+        if len(manifest_entries) == 0:
+            manifest_entries = self._read_submodules_from_commit(revision)
+        return manifest_entries
+
+    def _read_manifest_from_commit(self, revision) -> List[RepoEntry]:
+        proc = self._git_command("show", "{}:{}".format(revision, GitRepo.PKG_DEPENDENCY_FILE))
         if proc.returncode:
-            log.debug("No dependency file found in repo [{}:{}]".format(revision,
+            log.debug("No wit dependency file found in repo [{}:{}]".format(revision,
                       self.path))
             return []
         return RepoEntries.parse(proc.stdout, Path(GitRepo.PKG_DEPENDENCY_FILE), revision)
 
+    def _read_submodules_from_commit(self, revision) -> List[RepoEntry]:
+        proc = self._git_command("show", "{}:{}".format(revision, GitRepo.SUBMODULE_FILE))
+        if proc.returncode:
+            log.debug("No .gitmodule file found in repo [{}:{}]".format(revision, self.path))
+            return []
+
+        log.debug("{}:{} does not have wit-manifest.json, "
+                  "reading dependencies from git submodules instead"
+                  .format(self.name, revision))
+
+        # Use the 'git config' parser to read the submodule contents.
+        # Output is of the form:
+        #     submodule.$NAME.path $PATH
+        #     submodule.$NAME.url  $REMOTE
+        proc = self._git_command("config", "-f-", "--get-regex", "submodule..*",
+                                 input=proc.stdout)
+        self._git_check(proc)
+
+        paths_by_name = {}
+        urls_by_name = {}
+
+        path_r = re.compile('^submodule.(.*).path (.*)$')
```

Comment:
```suggestion
        path_r = re.compile(r'^submodule\.(.*)\.path (.*)$')
```

Same here with the regex.